cockatrice/src/gameselector.cpp

```diff
@@ -212,3 +216,17 @@ void GameSelector::processGameInfo(const ServerInfo_Game &info)
 {
     gameListModel->updateGameList(info);
 }
+
+void GameSelector::updateButtonChoices(const QModelIndex &index)
+{
+    if (!index.isValid())
+        return;
+
+    const ServerInfo_Game &game = gameListModel->getGame(index.data(Qt::UserRole).toInt());
+    bool overrideRestrictions = !tabSupervisor->getAdminLocked();
```

Comment:
The "supervisor" tab has a button that locks/unlocks admin superpowers.
The line is making the admin able to override the no-spectators-allowed constraint only when the superpowers are unlocked.